### PR TITLE
feat(SNAP_LIST): support of fetching specific snapshot details through SNAP_LIST

### LINKED ISSUE #2936: rebuild estimate for cstor replica
## Description
As of now CVR has a status field which describes replica health status and rebuilding information, i.e if it is rebuilding or not. But it doesn't say much about how much data it is missing from healthy replica or how much time it will take to become healthy.
If such information is added to CVR then it will help the users to get more idea about what is happening in rebuild process and the amount of time it will take to complete rebuilding.

## Possible Solution
CVR with existing/missing snapshot  details, plus rebuilding progress

### PR DESCRIPTION
**Pull request by mynktl**
fixes: https://github.com/openebs/openebs/issues/2936
Signed-off-by: mayank <mayank.patel@mayadata.io>

### CODE REVIEW

File: src/mgmt_conn.c [hunk 1]
```diff
@@ -1569,17 +1576,15 @@ process_message(uzfs_mgmt_conn_t *conn)
...
-				rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
-				    hdrp);
-				break;
-			}
+		snap = strchr(zvol_name, '@');
+		if (snap == NULL && hdrp->opcode != ZVOL_OPCODE_SNAP_LIST) {
+			LOG_ERR("Invalid snapshot name: %s", zvol_name);
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED, hdrp);
+			break;
+		} else if (snap != NULL) {
```
Comment (vishnuitta) [inline src/mgmt_conn.c]: I understand this code is correct.. but, for better readability and maintainability, lets decouple the cases here by removing 'else' and by just having 'if'

**Review by vishnuitta — COMMENTED** [review]
looks good.. one small change can be done if fine.. and testcases are missing

File: src/mgmt_conn.c (hunk 1 above)
Comment (mynktl) [reply src/mgmt_conn.c]: done

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**